packages/browser/src/Session.ts

```diff
@@ -129,6 +129,12 @@ export class Session extends EventEmitter {
    * @param init Optional parameters customizing the request, by specifying an HTTP method, headers, a body, etc. Follows the [WHATWG Fetch Standard](https://fetch.spec.whatwg.org/).
    */
   fetch = async (url: RequestInfo, init?: RequestInit): Promise<Response> => {
+    if (!this.info.isLoggedIn) {
+      // TODO: why does this.clientAuthentication.fetch return throws
+      // ""'fetch' called on an object that does not implement interface Window"
+      // when unauthenticated ?
```

Comment:
Let me guess, do we somewhere have `clientAuthentication.fetch = window.fetch`? That would explain it. It should be `window.fetch.bind(window)`.

Reply:
Well there is, but it's only on the `logout` code path. When logging in, the used fetch function is
```
const fetchWithCookies = (
  info: RequestInfo,
  init?: RequestInit
): ReturnType<typeof fetch> => {
  return window.fetch(info, {
    ...init,
    credentials: "include",
  });
};
```, 
because (for now) we rely on a RS cookie not to get logged out on refresh. So we do override a parameter, but `window.fetch` isn't rebound... Would you have a pointer to a read that would explain what your solution solves, so that I can see how it could apply here ?